bbq/vm/vm.go

```diff
@@ -1226,10 +1226,6 @@ func opFailableCast(vm *VM, ins opcode.InstructionFailableCast) {
 		// The failable cast may upcast to an optional type, e.g. `1 as? Int?`, so box
 		result = ConvertAndBox(vm.context, value, valueType, targetType)
 
-		// TODO:
-		// Failable casting is a resource invalidation
-		//interpreter.invalidateResource(value)
```

Comment:
This defensive check based on local "variables" is only applicable to the interpreter.